### PR TITLE
Compute Index.is_monotonic/Index.is_monotonic_decreasing in a distributed manner

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR proposes to compute `is_monotonic`/`is_monotonic_decreasing` in a distributed manner.

The approach is:

1. Check if it's ordered within each partition
2. Calculate min/max from each partition and check if the min/max of partitions are ordered.

### CODE REVIEW

File: databricks/koalas/indexing.py [hunk 1]
```diff
@@ -568,30 +568,19 @@ def _select_rows(self, rows_sel):
...
                             cond.append(index_column._scol <= F.lit(start).cast(index_data_type))
                         else:
                             raise KeyError(rows_sel.start)
                     if stop is None and rows_sel.stop is not None:
                         stop = rows_sel.stop
-                        if inc is not False:
+                        if index_column.is_monotonic_increasing is not False:
                             cond.append(index_column._scol <= F.lit(stop).cast(index_data_type))
-                        elif dec is not False:
+                        elif index_column.is_monotonic_decreasings is not False:
```
Comment (HyukjinKwon) [inline databricks/koalas/indexing.py]: Maybe we should ideally calculate both increasing and decreasing in one job but I noticed the codes become too much complicated here.

**Comment by HyukjinKwon** [comment]
Basically, it checks:

1. If the data in each partition is ordered
2. If each partition's min/max are ordered.

**Comment by HyukjinKwon** [comment]
Can't reproduce the test failure in my local ..

File: databricks/koalas/indexing.py [hunk 2]
```diff
@@ -568,30 +568,19 @@ def _select_rows(self, rows_sel):
...
                             cond.append(index_column._scol <= F.lit(start).cast(index_data_type))
                         else:
                             raise KeyError(rows_sel.start)
                     if stop is None and rows_sel.stop is not None:
                         stop = rows_sel.stop
-                        if inc is not False:
+                        if index_column.is_monotonic_increasing is not False:
                             cond.append(index_column._scol <= F.lit(stop).cast(index_data_type))
-                        elif dec is not False:
+                        elif index_column.is_monotonic_decreasing is not False:
```
Comment (ueshin) [inline databricks/koalas/indexing.py]: How about:

```py
inc = index_column.is_monotonic_increasing
if inc:
   dec = index_column.is_monotonic_decreasing

if start is None and rows_sel.start is not None:
    start = rows_sel.start
    if inc:
        ...
    elif dec:
        ...

...
```

**Comment by itholic** [comment]
Nice work

**Comment by itholic** [comment]
LGTM

**Pull request merged**